crypto/bn/bn_exp.c

```diff
@@ -618,6 +618,15 @@ int BN_mod_exp_mont_consttime(BIGNUM *rr, const BIGNUM *a, const BIGNUM *p,
 
     top = m->top;
 
+    if (in_mont != NULL && BN_is_zero(&in_mont->N)) {
+        BNerr(BN_F_BN_MOD_EXP_MONT_CONSTTIME, ERR_R_PASSED_INVALID_ARGUMENT);
+        return 0;
+    }
+    if ((unsigned int)top > INT_MAX / sizeof(m->d[0]) / (1 << 8)) {
```

Comment:
Given this depends on `BN_window_bits_for_ctime_exponent_size`, perhaps this should be based on `BN_MAX_WINDOW_BITS_FOR_CTIME_EXPONENT_SIZE`.